aws-xray-agent/src/main/java/com/amazonaws/xray/agent/runtime/handlers/upstream/ServletHandler.java

```diff
@@ -108,11 +111,19 @@ public void handleResponse(Event event) {
      * @param transactionState The current XRay transactional state
      */
     private void populateHeaderToTransactionState(HttpNetworkProtocolRequestEvent requestEvent, XRayTransactionState transactionState) {
+        String clientIP = requestEvent.getLocalIPAddress();
```

Comment:
Why `getLocalIPAddress()` is being used here instead of `getRemoteIPAddress()`? Latter one [returns client IP](https://github.com/awslabs/disco/blob/00d3c27d2b45da6b9ac82742c0e0ac6e3e7c980e/disco-java-agent/disco-java-agent-api/src/main/java/software/amazon/disco/agent/event/HttpNetworkProtocolRequestEvent.java#L66).

`getRemoteIPAddress()` should be the fall back option, otherwise `X-Forwarded-For` header will be ignored most of the time if not always.
I think the logic should be:
* check whether `X-Forwarded-For` header is set, use that IP
* retrieve clients IP address using `getRemoteIPAddress()`

Reply:
Thanks for the notes, probably shouldn't have copied so blindly! Looks like we should be using `getRemoteIPAddress`. Also, the X-Ray Java SDK checks X-Forwarded-For first as you suggest, so I'll also do that here.